### PR TITLE
Set up automated testing to be run via GitHub Actions w/ tric

### PR DESCRIPTION
**Pull request by borkweb**
This PR enables using `tric` and GitHub Actions to run our automated tests.

### CODE REVIEW

File: src/Tribe/Main.php [hunk 1]
```diff
@@ -3933,24 +3933,27 @@ public function get_closest_event( $post, $mode = 'next' ) {
...
+			$args = [
+				'post__not_in'   => [ $post->ID ],
+				'orderby'        => [
+					'start_date_clause' => $order,
+					'ID'                => $order,
+				],
 				'posts_per_page' => 1,
-				'meta_query'     => array(
-					array(
+				'meta_query'     => [
```
Comment (tr1b0t) [inline src/Tribe/Main.php:3943]: :warning: Detected usage of meta_query, possible slow query.

<sub>hash: 232a3de5f7c7a93d75653e166fc6b252</sub>

File: src/Tribe/Repositories/Event.php [hunk 2]
```diff
@@ -1471,23 +1471,28 @@ public function handle_order_by( $order_by ) {
...
 
 		foreach ( $check_orderby as $key => $value ) {
-			$loop++;
 			$order_by      = is_numeric( $key ) ? $value : $key;
-			$order         = is_numeric( $key ) ? 'ASC' : $value;
 			$default_order = Arr::get_in_any( [ $this->query_args, $this->default_args ], 'order', 'ASC' );
+			$order         = is_numeric( $key ) ? $default_order : $value;
+
+			// Let the first applied ORDER BY clause override the existing ones, then stack the ORDER BY clauses.
+			$override = $loop === 0;
```
Comment (tr1b0t) [inline src/Tribe/Repositories/Event.php:1479]: :no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: fb1bda9b5faff3f6e42ac5240d3049e1</sub>

File: src/Tribe/Repositories/Event.php [hunk 3]
```diff
@@ -1603,9 +1612,11 @@ protected function order_by_date( $use_utc, $after = false ) {
 			true
 		);
 
-		$order = Arr::get_in_any( [ $this->query_args, $this->default_args ], 'order', 'ASC' );
-		$this->filter_query->orderby( [ $meta_alias => $order ], $filter_id, true, $after );
-		$this->filter_query->fields( "MIN( {$postmeta_table}.meta_value ) AS {$meta_alias}", $filter_id, true );
+		$order = $order === null
```
Comment (tr1b0t) [inline src/Tribe/Repositories/Event.php:1615]: :no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: 17e290b8d1398bed3fbb2940dcb2278b</sub>

File: src/Tribe/Repositories/Event.php [hunk 4]
```diff
@@ -1641,10 +1656,12 @@ protected function order_by_organizer( $after = false ) {
 		);
 
 		$filter_id = 'order_by_organizer';
+		$order     = $order === null
```
Comment (tr1b0t) [inline src/Tribe/Repositories/Event.php:1659]: :no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: 259bf1967793249d0ae8ae50994157b7</sub>

File: src/Tribe/Repositories/Event.php [hunk 5]
```diff
@@ -1653,10 +1670,14 @@ protected function order_by_organizer( $after = false ) {
...
-	 * @param bool $after        Whether to append the order by clause to the ones managed by WordPress or not.
+	 * @param string $order      The order direction, either `ASC` or `DESC`; defaults to `null` to use the order
+	 *                           specified in the current query or default arguments.
+	 * @param bool   $after      Whether to append the order by clause to the ones managed by WordPress or not.
 	 *                           Defaults to `false`,to prepend them to the ones managed by WordPress.
+	 * @param bool   $override   Whether to override existing ORDER BY clauses with this one or not; default to
+	 *                           `true` to override existing ORDER BY clauses.
 	 */
-	protected function order_by_venue( $after = false ) {
+	protected function order_by_venue( $order = null,$after = false, $override = true ) {
```
Comment (tr1b0t) [inline src/Tribe/Repositories/Event.php:1680]: * :no_entry_sign: Expected 1 space between comma and argument "$after"; 0 found
* :no_entry_sign: Expected 1 space between comma and argument "$after"; 0 found

<sub>hash: b76879ee15fe268d36d0188694f0c036</sub>

File: src/Tribe/Repositories/Event.php [hunk 6]
```diff
@@ -1680,9 +1701,12 @@ protected function order_by_venue( $after = false ) {
 		);
 
 		$filter_id = 'order_by_venue';
-		$order     = Arr::get_in_any( [ $this->query_args, $this->default_args ], 'order', 'ASC' );
-		$this->filter_query->orderby( [ 'venue' => $order ], $filter_id, true, $after );
-		$this->filter_query->fields( "{$posts_table}.post_title AS venue", $filter_id, true );
+		$order     = $order === null
```
Comment (tr1b0t) [inline src/Tribe/Repositories/Event.php:1704]: :no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: 9e8ca05fff9f0ad74244a2a8c1c7d0a4</sub>

File: src/Tribe/Repositories/Event.php [hunk 7]
```diff
@@ -1721,4 +1745,49 @@ public function filter_by_hidden_on_upcoming( $hidden ) {
...
+						AND {$postmeta_table}.meta_key = %s
+					)
+				",
+				$meta_key
+			),
+			$filter_id,
+			true
+		);
+
+		$order = $order === null
```
Comment (tr1b0t) [inline src/Tribe/Repositories/Event.php:1787]: :no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: d0f5d40b17a6e4dd730f324db4e3722a</sub>

File: src/Tribe/Views/V2/Views/By_Day_View.php [hunk 8]
```diff
@@ -187,13 +187,24 @@ public function get_grid_days( $date = null, $force = false ) {
...
+			/*
+			 * Sort events in duration ascending order to make sure events that start on the same date and time
+			 * will be correctly positioned for multi-day, or all-day, parsing.
+			 * If not explicit, then events with the same start date and time would be sorted in the order MySQL
+			 * read them (not guaranteed).
+			 */
+			$order_by                   = tribe_normalize_orderby( $order_by, $order );
+			$order_by['event_duration'] = 'ASC';
+
+			$repo           = tribe_events()
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Views/By_Day_View.php:199]: :warning: Equals sign not aligned correctly; expected 1 space but found 11 spaces

<sub>hash: f6b249f865f0959614cdc7c0b145fe4c</sub>

**Comment by lucatume** [comment]
Closing this PR to start a new one with fewer merge conflicts based on `release/B20.08`.

**Pull request closed without merging**